integ-tests/src/cucumber/resources/features/dataservice/create_environment.feature

```diff
@@ -1,7 +1,15 @@
 @ignore
+@dataservice
 @create-environment
 Feature: Create environment
 
   Scenario: Create an environment
     When I create an environment
     Then the created environment response is valid
+
+  Scenario: Create an environment that already exists
+    Given I create an environment
+    When I try to create another environment with the same name
+    Then there should be an EnvironmentExistsException thrown
```

Comment:
This specification looks correct to me, and is what I would expect from the createEnvironment API call. However, as far as I can tell from the code, the `Environment` resource in the data service created by `createEnvironment` is the one that *can* have duplicate records for the same environment name, as long as they have different `environmentVersion` fields.

Reply:
The tests shouldn't match the code, the code should match what we think should be happening which is defined here :) What we need to add here is that the name cannot be duplicated within a cluster but can be duplicated across different clusters, I'll add a use case. And after the refactor of the environment and environmentRevision models what's described should happen in the APIs.

Reply:
> The tests shouldn't match the code, the code should match what we think should be happening which is defined here :)

100x :+1: